### DISCUSSION TITLE
Resource allocation for unfamiliar input sizes

### DISCUSSION THREAD

**Discussion by sgoldman101** [ld32/smarterSlurm#26] [General]
I'm wondering about resource allocation when an input file size lies outside the range of prior jobs recorded in jobRecords.txt.

For example, let's say a pipline processes 50M fastqs reads using 80G RAM and 8h but 150M reads use 140G RAM and 15h.  Assume that only 50M read samples have been run with smartSlurm so far and a new dataset of 150M reads/sample is being analyzed.  The large jobs will either run out of time or memory and automatically restart with incrementally higher specs.  My question is how many iterations of auto-restarting will it take before smartSlurm arrives as workable specs?

Could incrementing specs until the 150M read samples complete take days of reruns?  In cases like this, could the resource allocation be set up to deliberately overshoot after a certain number of attempts to ensure data analysis is not overly delayed waiting for resource allocation to step up?

Once smartSlurm has accumulated a comprehensive dataset for a specific tool using a wide range of inputs, this ceases to be an issue but seems like it might crop up for new users or new tools.

**Comment by ld32** [comment]
Good question. 
Right now, if a job has higher input size than all the jobs logged in jobRecord.txt, ssbatch uses the default resource (memory/runtime) defined in the bash pipeline script. If it is not define in the bash script, ssbatch uses the resource defined by the runAsPipeine command line. If the job runs out of resource, ssbatch will re-submit with more resource.
